modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/handler/AMImageRequestHandler.java

```diff
@@ -197,6 +288,51 @@ private Integer _getDistance(
 		return Math.abs(imageWidth - width);
 	}
 
+	private PermissionChecker _getPermissionChecker(
+			HttpServletRequest httpServletRequest)
+		throws Exception {
+
+		User user = _portal.getUser(httpServletRequest);
+
+		if (user == null) {
+			user = _userLocalService.getGuestUser(
+				_portal.getCompanyId(httpServletRequest));
+		}
+
+		return PermissionThreadLocal.getPermissionChecker(
+			user, !user.isGuestUser());
+	}
+
+	private User _getUser(HttpServletRequest httpServletRequest)
```

Comment:
Isn't this method redundant? You can simply get the permission checker, and from there get the user.